lib/hammer.ex

```diff
@@ -146,14 +187,31 @@ defmodule Hammer do
       end
   """
   def make_rate_checker(id_prefix, scale_ms, limit) do
-    fn (id) ->
-      check_rate("#{id_prefix}#{id}", scale_ms, limit)
-    end
+    make_rate_checker(:single, id_prefix, scale_ms, limit)
   end
 
-  defp call_backend(function, args) do
+  @spec make_rate_checker(backend::atom,
+                          id_prefix::String.t,
+                          scale_ms::integer,
+                          limit::integer)
+        :: ((id::String.t) -> {:allow, count::integer}
+                            | {:deny,  limit::integer}
+                            | {:error, reason::any})
+  @doc """
+  """
+  def make_rate_checker(backend, id_prefix, scale_ms, limit) do
+     fn (id) ->
+       check_rate(backend, "#{id_prefix}#{id}", scale_ms, limit)
+     end
+   end
+
+  defp call_backend(:single, function, args) do
```

Comment:
I'm wondering if we could pass the `:single/which` param all the way down to `get_backend_module`, and have it do The Right Thing based on it's input.